src/js/pages/Profile/index.jsx

```diff
@@ -11,13 +12,17 @@ import Offer from './components/Offer';
 import metadata from "../../features/metadata";
 import prices from "../../features/prices";
 import newBuy from "../../features/newBuy";
+import network from "../../features/network";
 
 import './index.scss';
 import Loading from "../../components/Loading";
 
 class Profile extends Component {
   componentDidMount() {
     this.props.load(this.props.match.params.address);
+    if(web3.utils.toChecksumAddress(this.props.match.params.address) === this.props.address){
```

Comment:
You should put this`if` above the load, and return so that we don't load for no reason